opentech/apply/review/views.py

```diff
@@ -21,11 +23,33 @@
 
 class ReviewContextMixin:
     def get_context_data(self, **kwargs):
-        staff_reviews = self.object.reviews.by_staff().select_related('author')
-        reviewer_reviews = self.object.reviews.by_reviewers().exclude(id__in=staff_reviews).select_related('author')
+        assigned = self.object.assigned.order_by('role__order').select_related('reviewer')
+        reviews = self.object.reviews.all().select_related('author')
+
+        reviews_block = defaultdict(list)
+        for assigned_reviewer in assigned:
+            reviewer = assigned_reviewer.reviewer
+            role = assigned_reviewer.role
+            review = reviews.filter(author=reviewer).first()
```

Comment:
We should not be doing this database query each iteration. Can you convert line 27 to something cached locally mapping user to review.

Reply:
Yes, I have cached it into a dictionary keyed by author `reviews_dict`